src/components/Navigation/index.js

```diff
@@ -0,0 +1,64 @@
+import PropTypes from "prop-types";
+import React, { Component } from "react";
+
+import "./index.css";
+
+class Navigation extends Component {
+  constructor(props) {
+    super(props);
+    this.state = { isActive: false };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick() {
+    this.setState(prevState => ({
+      isActive: !prevState.isActive
+    }));
+  }
+
+  render() {
+    const { isRevealing } = this.props;
+    const { isActive } = this.state;
+    const navigationClassName = `Navigation ${
+      isRevealing ? "Navigation--revealing" : ""
+    }`;
+    const detailsClassName = `Navigation-details ${
+      isActive ? "Navigation-details--active" : ""
+    }`;
+    return (
+      <nav className={navigationClassName}>
+        {isRevealing && (
+          <button className="Navigation-command" onClick={this.handleClick}>
```

Comment:
I have recently found out about the [`details`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details) tag which GitHub uses for their 'click to reveal' menus. Should we incorporate this practice?

Reply:
> Should we incorporate this practice?

There are limitations e.g. it's not possible to alter the animation. So, I went with a custom implementation (but used the "details" name).